### PR TITLE
Add envoy chart

### PR DESCRIPTION
**Pull request by cw-ozaki**
Created a helm chart for envoy.
This is because there are no other surviving charts.

#### Checklist

- [X] Chart Version bumped

### CODE REVIEW

**Comment by cw-ozaki** [comment]
The envoy configuration file is very flexible, so it does not include service and ingress linkage.

File: envoy/templates/hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{{- $root := . -}}
+  {{- if .Values.autoscaling.enabled -}}
```
Comment (cw-sakamoto) [inline envoy/templates/hpa.yaml:2]: unnecessary indent?

File: envoy/templates/hpa.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  {{- end }}
+  maxReplicas: {{ .Values.autoscaling.maxReplicas }}
+  metrics:
+    {{- toYaml .Values.autoscaling.metrics | nindent 4 }}
+  minReplicas: {{ .Values.autoscaling.minReplicas }}
+  scaleTargetRef:
+    kind: Deployment
+    apiVersion: apps/v1
+    name: "{{ template "envoy.fullname" . }}"
+  {{- end }}
```
Comment (cw-sakamoto) [inline envoy/templates/hpa.yaml:27]: same https://github.com/chatwork/charts/pull/237/files#r710725454

File: envoy/templates/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "envoy.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```
Comment (cw-sakamoto) [inline envoy/templates/ingress.yaml:8]: The beta1 of ingress is be removed in 1.22, and I don't think we need apiversion for `extensions,` as upstream does not support apiversion for `extenstions`.

https://kubernetes.io/blog/2021/08/04/kubernetes-1-22-release-announcement/

File: envoy/templates/pdb.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{{- if .Values.podDisruptionBudget.enabled }}
+apiVersion: policy/v1beta1
```
Comment (cw-sakamoto) [inline envoy/templates/pdb.yaml:2]: PDB is also depcated in v1beta1, so it should be supported.
https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/2136

File: envoy/templates/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  rules:
+    {{- range .Values.ingress.hosts }}
+    - host: {{ .host | quote }}
+      http:
+        paths:
+          {{- range .paths }}
+          - path: {{ .path }}
+            backend:
+              serviceName: {{ $fullName }}
+              servicePort: {{ $svcPort }}
```
Comment (cw-sakamoto) [inline envoy/templates/ingress.yaml:42]: @cw-ozaki 

The way ingress is written is different between v1 and v1beta1 😢
https://github.com/chatwork/charts/blob/master/akka/templates/ingress-http.yaml#L44-L52

Comment (cw-ozaki) [reply envoy/templates/ingress.yaml:42]: oops

**Review by cw-sakamoto — APPROVED** [review]

**Pull request merged**